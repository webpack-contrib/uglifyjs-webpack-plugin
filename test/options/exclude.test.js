import webpack from '../helpers/compiler';
import UglifyJsPlugin from '../../src/index';

describe('Options', () => {
  describe('exclude', () => {
    test('{RegExp}', async () => {
      const config = {
        entry: {
          1: `./exclude/1.js`,
          2: `./exclude/2.js`,
          entry: `./entry.js`,
        },
        plugins: [
          new UglifyJsPlugin({
            exclude: /1/,
          }),
        ],
      };

      const stats = await webpack(false, config);
      const { assets } = stats.compilation;

      expect(assets['1.bundle.js'].source()).toMatchSnapshot();
      expect(assets['2.bundle.js'].source()).toMatchSnapshot();
    });

    test('{Array<{RegExp}>}', async () => {
      const config = {
        entry: {
          1: `./exclude/1.js`,
          2: `./exclude/2.js`,
          entry: `./entry.js`,
        },
        plugins: [
          new UglifyJsPlugin({
            exclude: [/1/, /2/],
          }),
        ],
      };

      const stats = await webpack(false, config);
      const { assets } = stats.compilation;

      expect(assets['1.bundle.js'].source()).toMatchSnapshot();
      expect(assets['2.bundle.js'].source()).toMatchSnapshot();
    });
  });
});
