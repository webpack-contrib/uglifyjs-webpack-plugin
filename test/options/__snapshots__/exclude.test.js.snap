// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options exclude {Array<{RegExp}>} 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports) {

module.exports = function Bar1() {
  const b = 2 + 2;
  console.log(b + 1 + 2);
};

/***/ })
],[[0,3,0]]]);
//# sourceMappingURL=1.bundle.js.map"
`;

exports[`Options exclude {Array<{RegExp}>} 2`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[1],[
/* 0 */,
/* 1 */
/***/ (function(module, exports) {

module.exports = function Bar2() {
  const b = 2 + 2;
  console.log(b + 1 + 2);
};

/***/ })
],[[1,3,1]]]);
//# sourceMappingURL=2.bundle.js.map"
`;

exports[`Options exclude {RegExp} 1`] = `
"(window[\\"webpackJsonp\\"] = window[\\"webpackJsonp\\"] || []).push([[0],[
/* 0 */
/***/ (function(module, exports) {

module.exports = function Bar1() {
  const b = 2 + 2;
  console.log(b + 1 + 2);
};

/***/ })
],[[0,3,0]]]);
//# sourceMappingURL=1.bundle.js.map"
`;

exports[`Options exclude {RegExp} 2`] = `"(window.webpackJsonp=window.webpackJsonp||[]).push([[1],[,function(o,n){o.exports=function(){console.log(7)}}],[[1,3,1]]]);"`;
